### ISSUE TITLE
False negative on enforce-canonical-classes

### ISSUE THREAD

**Issue by kylemh** [schoero/eslint-plugin-readable-tailwind#369]
### Description

Seeing:
```The class `-mt-[0.04in]` can be written as `mt-[-3.84px]`(suggestCanonicalClasses)```

Firstly, this is in a document being printed, so `in` is on purpose; however, it's objectively NOT a simplification, so this one felt like a definite issue.

Also, `mt-[-0.04in]` does not trigger the same lint rule so it feels like a mistake all around.

### Flavor

JSX

### Code Input

```tsx
-mt-[0.04in]
```

### Expected Behavior

```tsx
No lint error.
```

### Actual Behavior

```tsx
The class `-mt-[0.04in]` can be written as `mt-[-3.84px]`(suggestCanonicalClasses)
```

### ESLint Log

```shell
/Users/kyle/Code/@company/ui/src/components/pharmacy-collations/order-label/PharmacyCollationMedicationManifest.tsx
  61:25  warning  The class: "-mt-[0.04in]" can be simplified to "mt-[-3.84px]"  better-tailwindcss/enforce-canonical-classes
```

### ESLint Config

```typescript
it's a doozey. i doubt it's relevant
```

### Versions

- ESLint: 10.2.1
- Parser: @typescript-eslint/parser@8.58.2
- Plugin: 4.4.1
- Node: 24.10.0
- TypeScript: 6.0.3

**Comment by schoero** [comment]
The logic what is converted is entirely in Tailwind CSS itself, so I can't change that behavior. 

What we could do at the ESLint plugin level is to add a list of ignore patterns, similar to the [`no-unknown-classes` rule](https://github.com/schoero/eslint-plugin-better-tailwindcss/blob/main/docs/rules/no-unknown-classes.md#ignore).

That way, you could create a regex that ignores classes that match arbitrary values with `in` units.

**Comment by kylemh** [comment]
I appreciate the workaround you're providing. I also opened https://github.com/tailwindlabs/tailwindcss-intellisense/issues/1573

**Comment by kylemh** [comment]
fwiw: https://github.com/tailwindlabs/tailwindcss/pull/19988
